bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -0,0 +1,281 @@
+			destination: Location,
+			message: Xcm<()>,
+			message_id: XcmHash,
+		},
+		/// Set OperatingMode
+		ExportOperatingModeChanged { mode: BasicOperatingMode },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Convert versioned location failure
+		UnsupportedLocationVersion,
+		/// Check location failure, should start from the dispatch origin as owner
+		InvalidAssetOwner,
+		/// Send xcm message failure
+		SendFailure,
+		/// Withdraw fee asset failure
+		FeesNotMet,
+		/// Convert to reanchored location failure
+		LocationConversionFailed,
+		/// Message export is halted
+		Halted,
+		/// The desired destination was unreachable, generally because there is a no way of routing
+		/// to it.
+		Unreachable,
+	}
+
+	impl<T: Config> From<SendError> for Error<T> {
+		fn from(e: SendError) -> Self {
+			match e {
+				SendError::Fees => Error::<T>::FeesNotMet,
+				SendError::NotApplicable => Error::<T>::Unreachable,
+				_ => Error::<T>::SendFailure,
+			}
+		}
+	}
+
+	/// The current operating mode for exporting to Ethereum.
+	#[pallet::storage]
+	#[pallet::getter(fn export_operating_mode)]
+	pub type ExportOperatingMode<T: Config> = StorageValue<_, BasicOperatingMode, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Initiates the registration for a Polkadot-native token as a wrapped ERC20 token on
+		/// Ethereum.
+		/// - `asset_id`: Location of the asset
+		/// - `metadata`: Metadata to include in the instantiated ERC20 contract on Ethereum
+		///
+		/// All origins are allowed, however `asset_id` must be a location nested within the origin
+		/// consensus system.
+		#[pallet::call_index(0)]
+		#[pallet::weight(
+			T::WeightInfo::register_token()
+				.saturating_add(T::BackendWeightInfo::transact_register_token())
+		)]
+		pub fn register_token(
+			origin: OriginFor<T>,
+			asset_id: Box<VersionedLocation>,
+			metadata: AssetMetadata,
+		) -> DispatchResult {
+			ensure!(!Self::export_operating_mode().is_halted(), Error::<T>::Halted);
+
+			let asset_location: Location =
+				(*asset_id).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
+			let origin_location = T::RegisterTokenOrigin::ensure_origin(origin, &asset_location)?;
+
+			let dest = T::BridgeHubLocation::get();
+			let call =
+				Self::build_register_token_call(&origin_location, &asset_location, metadata)?;
+			let remote_xcm = Self::build_remote_xcm(&call);
+			let message_id =
+				Self::send_xcm(origin_location.clone(), dest.clone(), remote_xcm.clone())
+					.map_err(|error| Error::<T>::from(error))?;
+
+			Self::deposit_event(Event::<T>::MessageSent {
+				origin: T::PalletLocation::get().into(),
+				destination: dest,
+				message: remote_xcm,
+				message_id,
+			});
+
+			Ok(())
+		}
+
+		/// Set the operating mode of the pallet, which can restrict messaging to Ethereum.
+		#[pallet::call_index(1)]
+		#[pallet::weight((T::DbWeight::get().reads_writes(1, 1), DispatchClass::Operational))]
+		pub fn set_operating_mode(
```

Comment:
@yrong 

Please move this function and its call index before that of `register_token`.

Also rename as follows:
```suggestion
		pub fn set_export_operating_mode(
```

Reply:
https://github.com/paritytech/polkadot-sdk/pull/7402/commits/63d7e5df474784fb417bad39ea5bc5d41559488b